Address sorting should include all addresses, including already attem… (#118)
tfpauly/draft-happy-eyeballs-v3: draft-ietf-happy-happyeyeballs-v3.md

change by either adding resolved addresses (for example due to DNS
push notifications {{?RFC8765}}) or removing previously resolved
addresses (for example, due to expiry of the TTL on that DNS record),
the client should react based on its current progress. Additionally, addresses
from SVCB IP hints SHOULD be removed from the list once A and AAAA records are
received for the corresponding name, if the addresses from the hints are absent from the
received records {{Section 7.3 of SVCB}}.

If an address is removed from the list that already had a connection
attempt started, the connection attempt SHOULD NOT be canceled, but
rather be allowed to continue. If the removed address had not yet
had a connection attempt started, it SHOULD be removed from the list
of addresses to try.

If an address is added to the list, its position SHOULD be determined by
applying the sorting rules (see {{sorting}}) to the complete list of addresses,

> including those already being attempted. This ensures that address family

Reviewer: It's not just address families that are being sorted. Can we use a more general term here?

> including those already being attempted. This ensures that address family

Reviewer 2: ```suggestion
including those previously received. This ensures that sorting rules, such as address family
```

> interleaving is maintained correctly regardless of when addresses arrive. For

Reviewer 2: ```suggestion
interleaving, are maintained correctly regardless of when addresses arrive. For
```

example, consider a connection attempt in which only IPv6 addresses
are available initially, and an attempt to one IPv6 address is already in progress.
Then, when IPv4 addresses are later received, an IPv4 address should be placed next in the list of addresses to attempt
(to account for interleaving address families) ahead of any remaining IPv6 addresses, as it would have

> been had it been available from the start.

Reviewer: ```suggestion
(to account for interleaving address families) ahead of any remaining IPv6 addresses, as if it had been available initially.
```
